### ISSUE TITLE
Add/Update Xcode 12.0 beta

### ISSUE THREAD

**Issue by ryanwilson** [actions/virtual-environments#1133]
**Tool information**
- Tool name: Xcode 12
- Tool license: https://www.apple.com/legal/sla/docs/xcode.pdf
- Add or update? Add
- Desired version: 12.0 beta 1
- Approximate size: ~15GB
- If this is an add request:
  - Brief description of tool: The latest beta of Apple's Xcode IDE
  - URL for tool's homepage: https://developer.apple.com/xcode/

**Area for Triage**:
Apple

**Question, Bug, or Feature?**:
Feature

**Virtual environments affected**
- [X] macOS 10.15
- [ ] Ubuntu 16.04 LTS
- [ ] Ubuntu 18.04 LTS
- [ ] Ubuntu 20.04 LTS
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Can this tool be installed during the build?**
I'm not entirely sure.

**Tool installation time in runtime**
I'm not entirely sure.

**Are you willing to submit a PR?**
I can try my best, would likely need some guidance though.

**Comment by maxim-lobanov** [comment]
Hello, we have already started to work on this request. We are planning to deploy new image with Xcode 12 at the middle next week (if there is no any issues)

**Comment by ryanwilson** [comment]
Awesome, thanks for the update! 😄 very happy to hear. Didn't see an open issue so thought I'd create one just in case.

**Comment by lewisdonofrio** [comment]
is anyone working on xcode IDE for ubuntu?  I mean if you want to be cross-platform, be cross platform ;)

**Comment by maxim-lobanov** [comment]
@lewisdonofrio , Currently, Xcode doesn't support other platforms except MacOS. Feel free to create feature request for Apple

**Comment by MaxDesiatov** [comment]
@lewisdonofrio if you're referring to a Swift IDE here, you can use Visual Studio code with [SourceKit-LSP](https://marketplace.visualstudio.com/items?itemName=pvasek.sourcekit-lsp--dev-unofficial) to get auto-complete working.

**Comment by markavitale** [comment]
@maxim-lobanov is the current plan to support/add both the Xcode 12 beta (build 12A6159) and the Xcode 12 for macOS Universal Apps beta (build 12A8158a)? Our macOS pipelines will likely require the "macOS universal apps beta" to start testing ARM support.

**Comment by maxim-lobanov** [comment]
@markavitale , No plans for adding `Xcode 12 for macOS Universal Apps beta (build 12A8158a)` for now. As I can see, It requires macOS 11.0 beta to run but we don't have plan to add new MacOS for now. Usually, we add new versions of macOS only after stable release.

**Comment by maxim-lobanov** [comment]
Hello, new image has been deployed and [Xcode 12 is available](https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.15-Readme.md#xcode) 🚀. 

We are planning to deliver all next beta versions of Xcode 12.0 in future.

**Comment by markavitale** [comment]
> @markavitale , No plans for adding `Xcode 12 for macOS Universal Apps beta (build 12A8158a)` for now. As I can see, It requires macOS 11.0 beta to run but we don't have plan to add new MacOS for now. Usually, we add new versions of macOS only after stable release.

@maxim-lobanov Xcode 12 for macOS Universal Apps beta does actually run on macOS 10.15 (Catalina). It allows you to build "Universal 2" binaries that contain both ARM and Intel implementations of a Mac app. It would presumably need to be deployed side-by-side with a different path, like `/Applications/Xcode_12_universal_beta.app`. While you can't actually run the ARM implementations on Catalina/current hardware, you can build your universal binaries on Catalina and then test them on an ARM developer transition kit on macOS 11.0 Big Sur.

Should I log a separate issue for getting the Xcode 12 universal beta included in a future image?

**Comment by paulb777** [comment]
I'm not able to find docs on how to select a specific Xcode version. Perhaps add a link here so that people searching for how to update an action to use Xcode 12 will find it?

**Comment by olejnjak** [comment]
> I'm not able to find docs on how to select a specific Xcode version. Perhaps add a link here so that people searching for how to update an action to use Xcode 12 will find it?

It is just a simple `sudo xcode-select -s {path_to_xcode}` I'm sure that everyone had used it some time during iOS/macOS development 🙂

**Comment by maxim-lobanov** [comment]
@markavitale , Thanks, it looks like I have mixed up something. Yes, please log the separate issue for `Xcode 12 for macOS Universal Apps beta`

**Comment by maxim-lobanov** [comment]
@paulb777 , you can simply use `sudo xcode-select -s /Applications/Xcode_12.app` command in your pipeline. Also you can use [action](https://github.com/marketplace/actions/setup-xcode-version) to switch between pre-installed Xcode versions so you won't need to remember commands and paths :)

**Comment by kiliankoe** [comment]
@maxim-lobanov Could it be that the path to Xcode 12 was updated? https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.15-Readme.md#xcode still lists it as /Applications/Xcode_12_beta.app, but it appears to be just /Applications/Xcode_12.app now? Just like in your last comment.

I opened https://github.com/actions/virtual-environments/pull/1173 to update that in the macOS readme.

**Comment by rogerluan** [comment]
> @maxim-lobanov Could it be that the path to Xcode 12 was updated? https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.15-Readme.md#xcode still lists it as /Applications/Xcode_12_beta.app, but it appears to be just /Applications/Xcode_12.app now? Just like in your last comment.
> 
> I opened #1173 to update that in the macOS readme.

I can attest this.

```diff
- sudo xcode-select -s /Applications/Xcode_12_beta.app/Contents/Developer
+ sudo xcode-select -s /Applications/Xcode_12.app/Contents/Developer
```

☝️ This change fixed my workflow.

**Comment by jmjauer** [comment]
> @markavitale , No plans for adding `Xcode 12 for macOS Universal Apps beta (build 12A8158a)` for now. As I can see, It requires macOS 11.0 beta to run but we don't have plan to add new MacOS for now. Usually, we add new versions of macOS only after stable release.

Any chance you add macOS 11 when it goes to public beta?

**Comment by maxim-lobanov** [comment]
@jmjauer , Hello, we don't have plans to add macOS 11 beta (usually, we add new macOS only after stable release due to maintenance concerns). However, could you please create an issue for this request? It will help us to track popularity of this request.
cc: @alepauly
